enhancements/single-node-production-deployment-approach.md

```diff
@@ -0,0 +1,484 @@
+need to disable the machine-approver-operator. An outside tool can be
+used to approve any certificate signing requests instead.
+
+#### Lack of high-availability
+
+In a single-node configuration OpenShift would have limited ability to
+provide resiliency features. We will need to clearly communicate these
+limitations to customers choosing to use single-node deployments.
+
+#### Failure to reboot
+
+It is possible that a single node may fail to reboot due to an issue
+with configuration, hardware, or environment. These single-node
+deployments will be managed by external tools, and as with other
+failures related to providing resilient operation, host boot failures
+will need to be detected and mitigated by those tools, rather than the
+local OpenShift instance.
+
+#### Reboots for reconfiguration
+
+Some reconfiguration operations will unavoidably cause the node to
+reboot. These tasks will need to be scheduled during maintenance
+windows.  Ideally we will be able to rely on the [graceful shutdown
+support in
+kubelet](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2000-graceful-node-shutdown)
+to stop all workloads safely as part of the reboot.
+
+## Design Details
+
+### Open Questions
+
+1. Telco workloads typically require special network setups
+   for a host to boot, including bonded interfaces, access to multiple
+   VLANs, and static IPs. How do we anticipate configuring those?
+2. How do we hand off ownership of the `etcd-quorum-guard` Deployment
+   from the `cluster-version-operator` to the `cluster-etcd-operator`?
+
+### Test Plan
+
+In order to claim full support for this configuration, we must have CI
+coverage informing the release. An end-to-end job using a single-node
+deployment and running an appropriate subset of the standard OpenShift
+tests will be created and configured to block accepting release images
+unless it passes.
+
+That end-to-end job should also be run against pull requests for the
+operators and other components that are most affected by the new
+configuration, such as the etcd and auth operators.
+
+There are 2 new CI jobs for testing the results. The
+e2e-metal-single-node-live-iso job
+(https://github.com/openshift/release/pull/14552) tests using the
+bootstrap-in-place approach described in
+https://github.com/openshift/enhancements/pull/565 on Packet and
+e2e-aws-single-node (https://github.com/openshift/release/pull/14556)
+uses AWS to test single-node deployments with a bootstrap VM.
+
+### Graduation Criteria
+
+##### Dev Preview
+
+- Ability to deploy an instance of OpenShift on a single host
+- Single-node deployments pass the identified conformance test suite(s)
+
+##### Dev Preview -> Tech Preview
+
+- Gather feedback from users rather than just developers
+- Multi-cluster management support for creating single-node
+  deployments. The work for that is well outside of the scope of this
+  enhancement.
+
+##### Tech Preview -> GA
+
```

Comment:
do we need any sort of longer running SN deployment before we release as GA - which I guess goes back to my old question of how long between tech preview & GA

Reply:
I don't think we're going to be able to commit to a specific period of time here. What we really want is a commitment that everyone will have "enough" time to gather input, and "enough" is going to depend quite a lot on what the early feedback is and how comfortable we all are with how things are working in real world scenarios. I feel like it's implicit that @smarterclayton , @eparis , and @derekwaynecarr will back us if we say the feature isn't ready for GA. None of us wants to ship something we're not confident in being able to support.

Who usually makes decisions about when features are ready for GA?

What if we add something like "team leads with implementation work for the enhancement sign-off" as a prerequisite to make it explicit?

Reply:
thinking a little more it's probably fair to revist this post-dev preview